File: src/app/Class/Song.ts
import { Album } from './Album';
import { Band } from './Band';
import { Genre } from './Genre';

export class Song {
  id: number;
  songName: string;
  trackNumber: number;
  songLengthSeconds: number;
  album: Album;
  band: Band;
  genre: Genre;
  songLyrics: string;
  musicVideoUrl: string;

  constructor(obj?: any) {
    this.id = (obj && obj.id) || 0;
    this.songName = (obj && obj.songName) || '';
    this.trackNumber = (obj && obj.trackNumber) || 0;
    this.songLengthSeconds = (obj && obj.songLengthSeconds) || 0;
    this.album = (obj && obj.album) || null;
    this.band = (obj && obj.band) || null;
    this.genre = (obj && obj.genre) || null;
    this.songLyrics = (obj && obj.songLyrics) || '';
    this.musicVideoUrl = (obj && obj.musicVideoUrl) || '';

  }

  // Getters

  getId(): number {
    return this.id;
  }

  getSongName(): string {
    return this.songName;
  }

  getTrackNumber(): number {
    return this.trackNumber;
  }

  getSongLengthSeconds(): number {
    return this.songLengthSeconds;
  }

  getSongLyrics(): string {
    return this.songLyrics;
  }

  getMusicVideoUrl(): string {
    return this.musicVideoUrl;
  }

  getAlbum(): Album {
    return this.album;
  }

  getBand(): Band {
    return this.band;
  }

  getGenre(): Genre {
    return this.genre;
  }

  // Setters

  setSongName(songName: string) {
    this.songName = songName;
  }

  setTrackNumber(trackNumber: number) {
    this.trackNumber = trackNumber;
  }

  setSongLengthSeconds(songLengthSeconds: number) {
    this.songLengthSeconds = songLengthSeconds;
  }

  setMusicVideoUrl(musicVideoUrl: string) {
    this.musicVideoUrl = musicVideoUrl;
  }

  setAlbum(album: Album) {
    this.album = album;
  }

  setBand(band: Band) {
    this.band = band;
  }

  setGenre(genre: Genre) {
    this.genre = genre;
  }

  setSongLyrics(songLyrics: string) {
    this.songLyrics = songLyrics;
  }

}

File: src/main/java/musiccube/entities/Song.java
package musiccube.entities;

import com.fasterxml.jackson.annotation.JsonInclude;

import javax.persistence.*;

@Entity
@JsonInclude(JsonInclude.Include.NON_NULL)
public class Song {

    @Id
    @GeneratedValue
    @Column
    private int id;

    @Column
    private String songName;

    @Column
    private int trackNumber;

    @Column
    private int songLengthSeconds;

    @ManyToOne
    private Album album;

    @ManyToOne
    private Band band;

    @ManyToOne
    private Genre genre;

    @Column
    private String songLyrics;

    public Song() {}

    public Song(String songName, int trackNumber, int songLengthSeconds, Album album, Band band, Genre genre, String songLyrics) {
        this.songName = songName;
        this.trackNumber = trackNumber;
        this.songLengthSeconds = songLengthSeconds;
        this.album = album;
        this.band = band;
        this.genre = genre;
        this.songLyrics = songLyrics;
    }

    public String getSongLyrics() {
        return songLyrics;
    }

    public void setSongLyrics(String songLyrics) {
        this.songLyrics = songLyrics;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getSongName() {
        return songName;
    }

    public void setSongName(String songName) {
        this.songName = songName;
    }

    public int getTrackNumber() {
        return trackNumber;
    }

    public void setTrackNumber(int trackNumber) {
        this.trackNumber = trackNumber;
    }

    public int getSongLengthSeconds() {
        return songLengthSeconds;
    }

    public void setSongLengthSeconds(int songLengthSeconds) {
        this.songLengthSeconds = songLengthSeconds;
    }

    public Album getAlbum() {
        return album;
    }

    public void setAlbum(Album album) {
        this.album = album;
    }

    public Band getBand() {
        return band;
    }

    public void setBand(Band band) {
        this.band = band;
    }


    public Genre getGenre() {
        return genre;
    }

    public void setGenre(Genre genre) {
        this.genre = genre;
    }

}